KNX troubleshooting (#17704)
home-assistant/home-assistant.github.io: source/_integrations/knx.markdown

<div class='note'>

> Unlike most KNX devices, Home Assistant defines for covers 0% is closed and 100% is fully open. The conversion is done internally by the KNX integration.

Reviewer: ```suggestion
Unlike most KNX devices, Home Assistant defines 0% as closed and 100% as fully open in regards to covers. The corresponding value inversion is done internally by the KNX integration.
```

  required: false
  type: [string, list]
angle_address:
  description: KNX group address for tilting the cover to the dedicated angle. *DPT 5.001*
  required: false
  type: [string, list]
angle_state_address:

>   description: Separate KNX group address for requesting the current tilt angle of cover. *DPT 5.001*

Reviewer: ```suggestion
  description: Separate KNX group address for requesting the current tilt angle of the cover. *DPT 5.001*
```

  type: boolean
  default: true
{% endconfiguration %}

## Troubleshooting / Common issues

### Logs for the KNX integration

`xknx`, the library used for KNX communication, provides various logging handlers for monitoring and debug purposes.
Add the following lines to your Home Assistant `configuration.yaml` to activate them:

```yaml
logger:
  default: warning
  logs:

>     # For most debugging needs `xnx.log` and one of `xknx.knx` or `xknx.telegram` be a good choice.

Reviewer: ```suggestion
    # For most debugging needs `xnx.log` and one of `xknx.knx` or `xknx.telegram` are a good choice.
```

    xknx: debug  # sets the level of all loggers
    xknx.log: debug  # provides general information (connection, etc.)
    xknx.raw_socket: debug  # logs incoming CEMI frames in raw hex format
    xknx.knx: debug  # logs incoming and outgoing KNX/IP frames at socket level
    xknx.telegram: debug  # logs telegrams before they are being processed at device level or sent to an interface
    xknx.state_updater: debug  # provides information about the state updater
```

> You can use the service logger.set_level to change log level of a handler on a running instance.

Reviewer: ```suggestion
You can use the service `logger.set_level` to change the log level of a handler on a running instance.
```

[![Open your Home Assistant instance and show your service developer tools.](https://my.home-assistant.io/badges/developer_services.svg)](https://my.home-assistant.io/redirect/developer_services/)

### Group address can not be read

> Every `*_state_address` is read on startup and sequentially if not configured differently. If you see the following errors in your logs a group address could not be read by a GroupValueRead request from Home Assistant in time.

Reviewer: ```suggestion
Every `*_state_address` is read on startup sequentially if not configured differently. If you see the following errors in your log, a group address could not be read by a GroupValueRead request from Home Assistant in time.
```

```log
> Could not sync group address '1/2/3' (Entity name - Feature)
> Error: KNX bus did not respond in time (2.0 secs) to GroupValueRead request for: '1/2/3'
```

#### No communication object (CO) assigned to the group address (GA) has the Read-Flag set in ETS

> - Enable the read flag for *one* CO assigned to the GA. Possibly the one most likely to hold the current state (e.g., for climates `brightness_state_address` the according CO of the dimming actuator).

Reviewer: ```suggestion
- Enable the read flag for *one* CO assigned to the GA. Use the one most likely to hold the current state (e.g., for a climate entity's `brightness_state_address` the according CO of the dimming actuator).
```

> #### Response Telegrams are not passing a line coupler, router or other filter in the installation

Reviewer: ```suggestion
#### Response telegrams are not passing a line coupler, router or other filter in the installation
```

- Use a dummy device in ETS for Home Assistant. These can be found in the ETS online catalog. Assign it to the line your interface connects Home Assistant to and link its communication objects to the group addresses you are using in Home Assistant. ETS will generate filter tables that are applied to your line couplers after updating their application.

#### Unresponsive system

- The timeout for logging the errors (2 seconds) is started when the GroupValueRead request is scheduled to be sent. On systems experiencing high loads sending can be delayed (e.g. Raspberry Pi running lots of integrations at startup).
Incoming response telegrams are always processed, so no information gets lost.

### Duplicate entities

If you find following error in your log you seem to have a duplicated entity in your configuration.

```log

> Platform knx does not generate unique IDs. ID 1/2/3 already exists - ignoring paltform.name

Reviewer: ```suggestion
Platform knx does not generate unique IDs. ID 1/2/3 already exists - ignoring platform.name
```

```

> `unique_id` for KNX entities is generated of required configuration values.

Reviewer: ```suggestion
The `unique_id` for KNX entities is generated based on required configuration values.
```

- binary_sensor: `state_address`
- climate: `temperature_address` `target_temperature_state_address` `target_temperature_address` `setpoint_shift_address`
- cover: `move_long_address` `position_address`
- fan: `address`

> - light: `address` or all `brightness_address` of `individual_colors`

Reviewer: ```suggestion
- light: `address` or all combined `brightness_address` if `individual_colors` is used
```

- notify: `address`
- scene: `address` and `scene_number`
- sensor: `state_address`
- switch: `address`
- weather: `address_temperature`

There can not be multiple entities on the same platform sharing these exact group addresses, even if they differ in other configuration.

### xknx.yaml configuration

> The configuration schema used in the YAML file configured in `config_file:` is deprecated since Home Assistant 2021.4. You can use the [xknx.yaml config converter](https://xknx.io/config-converter/) to convert it to a Home Assistant compatible `configuration.yaml` schema.

Reviewer: ```suggestion
The feature to specify a xknx configuration schema file in the Home Assistant configuration YAML file (via `config_file:`) is deprecated since Home Assistant 2021.4. You can use the [xknx.yaml config converter](https://xknx.io/config-converter/) to convert it to a Home Assistant compatible `configuration.yaml` schema.
```

  type: boolean
  default: true
{% endconfiguration %}

## Troubleshooting / Common issues

### Logs for the KNX integration

`xknx`, the library used for KNX communication, provides various logging handlers for monitoring and debug purposes.
Add the following lines to your Home Assistant `configuration.yaml` to activate them:

```yaml
logger:
  default: warning
  logs:
    # For most debugging needs `xnx.log` and one of `xknx.knx` or `xknx.telegram` are a good choice.
    xknx: debug  # sets the level of all loggers
    xknx.log: debug  # provides general information (connection, etc.)
    xknx.raw_socket: debug  # logs incoming UDP frames in raw hex format
    xknx.knx: debug  # logs incoming and outgoing KNX/IP frames at socket level
    xknx.telegram: debug  # logs telegrams before they are being processed at device level or sent to an interface
    xknx.state_updater: debug  # provides information about the state updater
```

You can use the service `logger.set_level` to change the log level of a handler on a running instance.

> {% my developer_services badge %}

Reviewer 2: ```suggestion
{% my developer_call_service badge service="logger.set_level" %}
```

Reviewer 2: @farmio Yes 😉  (partly)

> [![Open your Home Assistant instance and show your service developer tools.](https://my.home-assistant.io/badges/developer_services.svg)](https://my.home-assistant.io/redirect/developer_services/)

Reviewer 2: ```suggestion
{% my developer_services badge %}
```

Author: 😃 cool!
Is it possible to also prefill the service data parameters?

Every `*_state_address` is read on startup sequentially if not configured differently. If you see the following errors in your log, a group address could not be read by a GroupValueRead request from Home Assistant in time.

```log
> Could not sync group address '1/2/3' (Entity name - Feature)
> Error: KNX bus did not respond in time (2.0 secs) to GroupValueRead request for: '1/2/3'
```

#### No communication object (CO) assigned to the group address (GA) has the Read-Flag set in ETS

- Enable the read flag for *one* CO assigned to the GA. Use the one most likely to hold the current state (e.g., for a light entity's `brightness_state_address` the according CO of the dimming actuator).

#### Response telegrams are not passing a line coupler, router or other filter in the installation

- Use a dummy device in ETS for Home Assistant. These can be found in the ETS online catalog. Assign it to the line your interface connects Home Assistant to and link its communication objects to the group addresses you are using in Home Assistant. ETS will generate filter tables that are applied to your line couplers after updating their application.

#### Unresponsive system

> - The timeout for logging the errors (2 seconds) is started when the GroupValueRead request is scheduled to be sent. On systems experiencing high loads sending can be delayed (e.g. Raspberry Pi running lots of integrations at startup).
> Incoming response telegrams are always processed, so no information gets lost.

Reviewer 2: ```suggestion
- The timeout for logging the errors (2 seconds) is started when the GroupValueRead request is scheduled to be sent. On systems experiencing high loads sending can be delayed (e.g., Raspberry Pi running lots of integrations at startup).
Incoming response telegrams are always processed, so no information gets lost.
```
